syntax = "proto3";

service BirdWiki {

  rpc listBirds(BirdName) returns (stream BirdInfo) {}

  rpc getBird(BirdName) returns (BirdInfo) {}

  rpc readBird(BirdName) returns (BirdPage) {}
  
  rpc editBird(BirdName) returns (BirdPage) {}

  rpc saveBird(BirdPage) returns (Confirmation) {}
}

service LoginUser {
    rpc login (UserLogin) returns (UserInfo) {}
}

message BirdName { string name = 1; }

message BirdInfo {
  string name = 1;
  bool editing = 2;
  string text = 4;
}

message BirdPage {
  string name = 1;
  string text = 2;
}

message Confirmation { bool saved = 1; }

message UserInfo {
    string crBio = 1;
    string name = 2;
    string password_hash = 3;
}

message UserLogin {
    string crBio = 1;
    string password = 2;
}